CromIAM/src/test/scala/cromiam/sam/SamClientSpec.scala

```diff
@@ -117,10 +120,80 @@ class SamClientSpec extends AsyncFlatSpec with Matchers with BeforeAndAfterAll
     }
   }
 
-  it should "return sam errors while checking user authorization" in {
-    samClientWithError.requestSubmission(unauthorizedUserWithNoCollection, unauthorizedCollection, emptyHttpRequest)
-      .value.unsafeToFuture() map {
-      _.left.value should be(expectedErrorResponse)
+  it should "add the user when create returns a conflict" in {
+    val samClient = new BaseMockSamClient() {
+      override protected def registerCreation(user: User,
+                                              collection: Collection,
+                                              cromiamRequest: HttpRequest): FailureResponseOrT[HttpResponse] = {
+        val conflictResponse = HttpResponse(StatusCodes.Conflict, entity = HttpEntity("expected conflict"))
+        returnResponse(conflictResponse)
+      }
+
+      override def requestAuth(authorizationRequest: CollectionAuthorizationRequest,
+                               cromiamRequest: HttpRequest): FailureResponseOrT[Unit] = {
+        Monad[FailureResponseOrT].unit
+      }
+    }
+    samClient
+      .requestSubmission(unauthorizedUserWithNoCollection, unauthorizedCollection, emptyHttpRequest)
+      .map(_ => succeed)
```

Comment:
But what happens if this never gets called? Does the test finish running and never fail, thus succeed by default?

Could you make this update a "I did what you expected" variable to true, and assert that it eventually happens with something like this?
```
eventually {
  didWhatIExpected should be(true)
}
```

Reply:
I'm assuming the "never gets called" refers to the `requestSubmission`/`requestAuth` loop-vs-create/update discussion. If there's no loop the code shouldn't need an eventually.

Reply:
I mean that even without this `.map(_ => succeed)`, or even if the operation fails, the test wouldn't this test still "pass"... because that's the default and nothing is ever asserting otherwise?

Reply:
This class is an [Asynchronous ScalaTest Spec](http://www.scalatest.org/user_guide/async_testing), required to return a `Future[Assertion]`<sup>1</sup>. That's the only reason why the `map(_ => succeed)` is even there, because the api/compiler won't allow you to even accidentally return anything else, not even `Unit` nor `Future[Unit]`.

So if the operation fails, the result will be a failed-Future, which will fail test.

<hr/>
<sup>1</sup> There is built-in ScalaTest sugar that allows one to also end an Async-Spec with just an `Assertion` like `succeed` that will implicitly get converted to `Future[Assertion]`.

Reply:
Aha! That's actually a cool pattern - and which I have probably reinvented at various points in the past. The more you know...